students/travis_nelson/lesson02/gridPrinter.py

```diff
@@ -0,0 +1,107 @@
+
+def grid():
+    """"Prints a 2x2 grid and returns a completion message."""
+    print('+', end=' ')
+    print('- '*4, end= '')
+    print('+', end=' ')
+    print('- '*4, end= '')
+    print('+')
+    print('|', end = '')
+    print(' '*8, end= ' ')
+    print('|', end = '')
+    print(' '*8, end= ' ')
+    print('|')
+    print('|', end = '')
+    print(' '*8, end= ' ')
+    print('|', end = '')
+    print(' '*8, end= ' ')
+    print('|')
+    print('|', end = '')
+    print(' '*8, end= ' ')
+    print('|', end = '')
+    print(' '*8, end= ' ')
+    print('|')
+    print('|', end = '')
+    print(' '*8, end= ' ')
+    print('|', end = '')
+    print(' '*8, end= ' ')
+    print('|')
+    print('+', end=' ')
+    print('- '*4, end= '')
+    print('+', end=' ')
+    print('- '*4, end= '')
+    print('+')
+    print('|', end = '')
+    print(' '*8, end= ' ')
+    print('|', end = ' ')
+    print(' '*8, end= '')
+    print('|')
+    print('|', end = '')
+    print(' '*8, end= ' ')
+    print('|', end = ' ')
+    print(' '*8, end= '')
+    print('|')
+    print('|', end = '')
+    print(' '*8, end= ' ')
+    print('|', end = '')
+    print(' '*8, end= ' ')
+    print('|')
+    print('|', end = ' ')
+    print(' '*8, end= '')
+    print('|', end = ' ')
+    print(' '*8, end= '')
+    print('|')
+    print('+', end=' ')
+    print('- '*4, end= '')
+    print('+', end=' ')
+    print('- '*4, end= '')
+    print('+')
+    return "Done"
+
+def print_grid(lengthOfCell=2):
+    """Prints a 2x2 grid of int parameter cell-length/width"""
+    def grid_columns(lengthOfCell):
+        """Prints the horizontal borders of the grid. The length of these borders is specified by the input parameter"""
+        print('+', end=' ')
+        print('- '*lengthOfCell, end= '')
+        print('+', end=' ')
+        print('- '*lengthOfCell, end= '')
+        print('+')
+    def grid_row(lengthOfCell):
+        """Prints a row of vertical grid borders"""
+        print('|', end = ' ')
+        print(' '*(lengthOfCell*2), end= '')
+        print('|', end = ' ')
+        print(' '*(lengthOfCell*2), end= '')
+        print('|')   
```

Comment:
although inner functions have its uses be careful about over-using them. functions here are best when defined as regular functions.